aptos-move/framework/supra-stdlib/sources/enumerable_map.move

```diff
@@ -122,57 +123,93 @@ module supra_std::enumerable_map {
+    public fun get_value<K: copy+drop, V: store+drop+copy>(map: & EnumerableMap<K, V>, key: K): V {
         table::borrow(&map.map, key).value
     }
 
     /// Returns reference to the value of a key that is present in Enumerable Map
-    public fun get_value_ref<K : copy+drop, V : store+drop+copy>(map: & EnumerableMap<K, V>, key: K): &V {
+    public fun get_value_ref<K: copy+drop, V: store+drop+copy>(map: & EnumerableMap<K, V>, key: K): &V {
         &table::borrow(&map.map, key).value
     }
 
 
     /// Returns the value of a key that is present in Enumerable Map
-    public fun get_value_mut<K : copy+drop, V : store+drop+copy>(map: &mut EnumerableMap<K, V>, key: K): &mut V {
+    public fun get_value_mut<K: copy+drop, V: store+drop+copy>(map: &mut EnumerableMap<K, V>, key: K): &mut V {
         &mut table::borrow_mut(&mut map.map, key).value
     }
 
     /// Returns the list of keys that the Enumerable Map contains
-    public fun get_map_list<K : copy+drop, V : store+drop+copy>(map: &EnumerableMap<K, V>): vector<K> {
+    public fun get_map_list<K: copy+drop, V: store+drop+copy>(map: &EnumerableMap<K, V>): vector<K> {
         return map.list
     }
 
     /// Check whether Key is present into the Enumerable map or not
-    public fun contains<K: copy+drop, V : store+drop+copy>(map: &EnumerableMap<K, V>, key: K): bool {
+    public fun contains<K: copy+drop, V: store+drop+copy>(map: &EnumerableMap<K, V>, key: K): bool {
         table::contains(&map.map, key)
     }
 
     /// Return current length of the EnumerableSetRing
-    public fun length<K : copy+drop, V : store+drop+copy>(set: &EnumerableMap<K, V>): u64 {
+    public fun length<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>): u64 {
         return vector::length(&set.list)
     }
 
+    /// Apply the function to each element in the EnumerableMap.
+    public inline fun for_each_value<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>, f: |V|) {
+        let i = 0;
+        let len = length(set);
+        while (i < len) {
+            let key = *vector::borrow(&set.list, i);
+            f(table::borrow(&set.map, key).value);
+            i = i + 1
+        }
+    }
+
+    /// Apply the function to a reference of each element in the EnumerableMap.
+    public inline fun for_each_value_ref<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>, f: |&V|) {
+        let i = 0;
+        let len = length(set);
+        while (i < len) {
+            let key = *vector::borrow(&set.list, i);
+            f(&table::borrow(&set.map, key).value);
+            i = i + 1
+        }
+    }
+
+    /// Filter the enumerableMap using the boolean function, removing all elements for which `p(e)` is not true.
+    public inline fun filter<K: copy+drop, V: store+drop+copy>(set: &EnumerableMap<K, V>, p: |&V|bool): vector<V> {
+        let result = vector<V>[];
+        for_each_value(set, |v| {
```

Comment:
I believe `for_each_value_ref` would be better here.

Reply:
We can use that; however, since the next step involves using the `vector::push_back` method, we need to handle pointer dereferencing with vector::push_back(&mut result, `*v`). Therefore, there doesn't seem to be a specific reason to use `for_each_value_ref` in this context.